tools/deploy_ipk.sh

```diff
@@ -1,16 +1,22 @@
 #!/bin/sh -e
+###############################################################
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+# Copyright (c) 2019 Raphaël Mélotte
+# Copyright (c) 2020 Tomer Eliyahu (Intel)
+# This code is subject to the terms of the BSD+Patent license.
+# See LICENSE file for more details.
+###############################################################
 
 usage() {
-    echo "usage: $(basename "$0") [-h]"
+    echo "usage: $(basename "$0") <user>@<target> <ipk> [-h]"
     echo "  options:"
     echo "      -h|--help - show this help menu"
     echo "      --certification-mode - enable certification-mode on the target"
+    echo "      --proxy - use ssh proxy <user>@<proxy>"
 }
 
 deploy() {
-    SSH_OPTIONS="-oBatchMode=yes"
-
-    if ! ssh "$SSH_OPTIONS" "$1" /bin/true ; then
+    if ! eval ssh "$SSH_OPTIONS" "$1" /bin/true ; then
```

Comment:
I often avoid using eval because I find it hard to get right. 
I know BatchMode was already in a variable so it's tempting to add more options to it, but it was a bad idea from the beginning.
How about using a function that always uses BatchMode, and that uses the proxy option depending if the option was used or not?

Reply:
I used `eval` based on a comment by @arnout about not using `sh -c`. I'm not sure what you mean with this proposal - can you provide a short example?

Reply:
I decided this has been waiting for far too long. We also need it for UGW since we have no other way to deploy to target there (need the deploy_ipk.sh script), so I'll merge this and keep this comment open so we can address this in a new PR.

Reply:
I don't know any clean way to dynamically build a list of options in POSIX sh.
It's either eval, or arrays, but arrays are not POSIX except for one exception: `$@`.
This means you could use it to pass the correct options to a new function for example.

Either you have to play with the arguments passed to the script to shape the array in such a way that it can directly be passed to ssh/scp (e.g. remove the arguments you know and give everything else to ssh), or you have to call a function with all the options you're interested in, some being empty, and filter out the empty ones in the function.

Maybe we could also require bash for this script, since it's meant to be used on development platforms, not on the targets directly.